src/main/host/syscall/handler/sysinfo.rs

```diff
@@ -0,0 +1,53 @@
+use crate::core::support::emulated_time;
+use crate::core::worker::Worker;
+use crate::host::context::ThreadContext;
+use crate::host::syscall::handler::SyscallHandler;
+use crate::host::syscall_types::{PluginPtr, SysCallArgs, SyscallResult, TypedPluginPtr};
+use crate::utility::pod;
+
+use nix::errno::Errno;
+use syscall_logger::log_syscall;
+
+impl SyscallHandler {
+    // Using libc::c_void here because the libc::sysinfo struct is a return value only.
+    #[log_syscall(/* rv */ libc::c_int, /* info */ *const libc::sysinfo)]
+    pub fn sysinfo(&self, ctx: &mut ThreadContext, args: &SysCallArgs) -> SyscallResult {
+        // Pointer to the plugin memory where we write the result.
+        let info_ptr = TypedPluginPtr::new::<libc::sysinfo>(PluginPtr::from(args.get(0)), 1);
+        if info_ptr.is_null() {
+            return Err(Errno::EFAULT.into());
+        }
+
+        // Seconds are needed for uptime.
+        let time = Worker::current_time().unwrap_or(emulated_time::SIMULATION_START);
+        let seconds = time
+            .duration_since(&emulated_time::SIMULATION_START)
+            .as_secs();
+
+        // Get a zeroed struct to make sure we init all fields.
+        let mut info = pod::zeroed::<libc::sysinfo>();
+
+        // These values are chosen arbitrarily; we don't think it matters too
+        // much, except to maintain determinism. For example, Tor make decisions
+        // about how many circuits to allow to be open (and other OOM settings)
+        // based on available memory.
+        info.uptime = i64::try_from(seconds).unwrap_or(0);
```

Comment:
Maybe change the `unwrap_or(0)` to `unwrap_or(i64::MAX)`?

Reply:
I was going to us `i64::MAX` but wasn't sure if that would be the only reason `try_from` would fail. But I think it's the most likely reason so I think it makes sense.